Add autonomous cruise controller with intersection handling

siddarthnandy opened the pull request:
- Implements Pure Pursuit lateral control for path following
- Implements Adaptive PID longitudinal control for speed management
- Handles traffic lights, stop signs, 4-way stops, and roundabouts
- Includes adaptive cruise control with safe following distance
- Optimized for GEM e6 vehicle specifications
- Provides multiple launch configurations
- Full documentation and parameter files included

Author: Siddarth Nandyala <siddarth.nandyala@utdallas.edu>
Organization: Nova UTD Autonomous Driving

Commit:
Add autonomous cruise controller with intersection handling

- Implements Pure Pursuit lateral control for path following
- Implements Adaptive PID longitudinal control for speed management
- Handles traffic lights, stop signs, 4-way stops, and roundabouts
- Includes adaptive cruise control with safe following distance
- Optimized for GEM e6 vehicle specifications
- Includes comprehensive unit tests (23 tests)
- Provides multiple launch configurations
- Full documentation and parameter files included

Author: Siddarth Nandyala <siddarth.nandyala@utdallas.edu>
Organization: Nova UTD Autonomous Driving

siddarthnandy committed:
Update autonomous_cruise_intersection_node.py

siddarthnandy committed:
Update autonomous_cruise_node.py

pranavboyapati commented on src/autonomous_cruise/launch/autonomous_cruise.launch.py:
Instead of creating a new launch file for this feature, we'd like for this node to be added to the existing launch file structure

pranavboyapati commented on src/autonomous_cruise/launch/gem_e6.launch.py:
Same thing here, let's add it to the existing launch file structure

pranavboyapati commented on src/autonomous_cruise/launch/gem_e6_intersection.launch.py:
Same thing here too, let's add this node to the existing launch file structure

pranavboyapati commented on param/autonomous_cruise_params.yaml:
The file itself is fine but lets move it into the existing param directory

pranavboyapati commented on param/gem_e6_intersection_params.yaml:
Again, this file looks good but should be moved to the param directory

pranavboyapati commented on param/gem_e6_params.yaml:
This should also be moved to the param directory

pranavboyapati commented on src/autonomous_cruise/autonomous_cruise/intersection_handler.py:
```diff
@@ -0,0 +1,324 @@
+    
+    def __init__(
+        self,
+        approach_speed: float = 3.0,
```
At least some of the values in lines 39-47 are in the params files. Can we set the variables to use values from the params file instead of hardcoding values here?

siddarthnandy commented:
Thanks for the review. I will make this updates

Commit:
Address PR review comments: integrate nodes into existing launch structure and move params to root directory

- Remove separate launch files from autonomous_cruise package
- Add autonomous_cruise_controller and autonomous_cruise_intersection_controller node definitions to launches/launch_node_definitions.py
- Move param files from src/autonomous_cruise/params/ to param/ directory
- Maintains all functionality while following existing repository structure

Author: Siddarth Nandyala <siddarth.nandyala@utdallas.edu>
